### PR TITLE
add codecov

### PR DESCRIPTION
**Pull request by rrlamichhane**
Please refer to [CONTRIBUTING.md](CONTRIBUTING.md) for detailed instructions.

## Summary

- Enable codecov
- Update python versions for CI containers
- Update lock file

## PR Checklist
- [ ] My PR is less than 500 lines of code
- [ ] I have added sufficient comment as docstrings in my code
- [ ] I have made corresponding changes to the documentation

- [ ] I have written unit-tests to test all of my code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [![CircleCI](https://dl.circleci.com/status-badge/img/gh/sambanova/generative_data_prep/tree/main.svg?style=svg)](https://dl.circleci.com/status-badge/redirect/gh/sambanova/generative_data_prep/tree/main)
+[![codecov](https://codecov.io/gh/sambanova/generative_data_prep/graph/badge.svg?token=9CYRCUOOAO)](https://codecov.io/gh/sambanova/generative_data_prep)
```
Comment (snova-zoltanc) [inline README.md:2]: image
Looks like this banner is not displaying properly?

**Review by snova-zoltanc — COMMENTED** [review]
This is awesome, thanks for adding code coverage!

I was looking through code coverage report and sometimes it is annoying like abstract methods are labeled as not being tested, and some edge case error messages are not being tested, is there any way to avoid these being labeled as not covered? 

Also some files I don't want to test, for example my custom shuffling implementation it is indeterministic so hard to test, can we ignore these files somehow ?

image

**Review by snova-zoltanc — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (rrlamichhane) [reply README.md:2]: That'll update once this merges to main since it shows the status of `main`

**Comment by rrlamichhane** [comment]
@snova-zoltanc I'm not sure if we can ignore that but I can look into it in the future.

Also, I don't think its too difficult to cover them, we just need to parametrize one pytest with all these callbacks and see if the appropriate exceptions are raise.

**Review by snova-kevinb — APPROVED** [review]

**Comment by snova-zoltanc** [comment]
> @snova-zoltanc I'm not sure if we can ignore that but I can look into it in the future.
> 
> Also, I don't think its too difficult to cover them, we just need to parametrize one pytest with all these callbacks and see if the appropriate exceptions are raise.

Sounds great, I may ask for some recommendations on this if we plan to increase code coverage in the future.

**Comment by rrlamichhane** [comment]
@snova-zoltanc sounds good; for sure! always happy to help when I can
